src/quemb/molbe/autofrag.py

```diff
@@ -107,14 +116,82 @@ def remove_nonnunique_frags(self, natm: int) -> None:
             if subsets:
                 sorted_subsets = sorted(subsets, reverse=True)
                 for bdx in sorted_subsets:
-                    del self.center[bdx]
-                    del self.fsites[bdx]
-                    del self.fs[bdx]
-                    del self.center_atoms[bdx]
-                    del self.fragment_atoms[bdx]
-
+                    if len(self.fsites) == 1:
+                        # If all fragments are identified as subsets,
+                        # this stops the loop from deleting the final fragment.
+                        break
+                    else:
+                        # Otherwise, delete the subset fragment.
+                        del self.center[bdx]
+                        del self.fsites[bdx]
+                        del self.fs[bdx]
+                        del self.center_atoms[bdx]
+                        del self.fragment_atoms[bdx]
+                        del self.edge_list[bdx]
         return None
 
+    def export_graph(
+        self,
+        outdir: PathLike,
+        outname: str = "AdjGraph",
+        cmap: str = "cubehelix",
+        node_position: str = "coordinates",
+    ) -> None:
+        outdir = Path(outdir)
+        F = 2
+        c_ = plt.cm.get_cmap(cmap)
+        c = [
+            c_(fdx / len(self.edge_list))[0:3] for fdx in range(0, len(self.edge_list))
+        ]
+        patches = [mpatches.Patch(color=color, alpha=0.9) for color in c]
+        labels = {adx: (map["label"] + str(adx)) for adx, map in self.adx_map.items()}
+
+        G = self.adjacency_graph
+
+        if node_position in ["coordinates"]:
+            pos = [
+                (
+                    map["coord"][0] + (map["coord"][2] / F),
+                    map["coord"][1] + (map["coord"][2] / F),
+                )
+                for _, map in self.adx_map.items()
```

Comment:
for _, map in self.adx_map.items()

is equivalent to

`for map in self.adx_map.values()`